core/kernel/tee_ta_manager.c

```diff
@@ -539,23 +538,25 @@ static TEE_Result tee_ta_init_session_with_context(struct tee_ta_session *s,
 		condvar_wait(&tee_ta_init_cv, &tee_ta_mutex);
 	}
 
-	/*
-	 * If TA isn't single instance it should be loaded as new
-	 * instance instead of doing anything with this instance.
-	 * So tell the caller that we didn't find the TA it the
-	 * caller will load a new instance.
-	 */
-	if ((ctx->flags & TA_FLAG_SINGLE_INSTANCE) == 0)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	if (is_user_ta_ctx(&ctx->ts_ctx)) {
```

Comment:
The flags `TA_FLAG_SINGLE_INSTANCE` and `TA_FLAG_MULTI_SESSION` apply to all TAs.
See for instance `PTA_MANDATORY_FLAGS`

Reply:
What is the point of checking if it's a user_ta first? Even if those two checks are always false for non-user_ta's, it is not harmful and those TAs must maintain those flags properly anyway.

Reply:
Right, indeed we could only test `ctx->flags`.

Regarding the debug trace message `"Re-open TA <uuid>"`, it was printed only for (user) TAs. I would be in favor of preserving this, to avaid confusion:
```c
if (is_user_ta_ctx(&ctx->ts_ctx))
	DMSG("Re-open TA %pUl", (void *)&ctx->ts_ctx.uuid);
```
or maybe change that trace message to make it more generic (TA + PTA + StMM). Maybe something like:
```c
DMSG("Re-open trusted service %pUl", (void *)&ctx->ts_ctx.uuid);
```

Reply:
The latter is fine.